server/item/armour_trim.go

```diff
@@ -0,0 +1,48 @@
+package item
+
+import "github.com/df-mc/dragonfly/server/world"
+
+type ArmourTrim struct {
+	Template ArmourTrimTemplate
+	Material TrimMaterial
+}
+
+type TrimMaterial interface {
+	// TrimMaterial returns the material name used for reading and writing trim data.
+	TrimMaterial() string
+	// MaterialColor returns the color code used for internal text formatting. Use text.Colourf for proper formatting.
+	MaterialColor() string
```

Comment:
Also please use British English when possible, so MaterialColour() rather than MaterialColor()

Reply:
Oops I did mean to use British English will fix